core/src/main/java/org/sonarsource/sonarlint/core/ConnectedSonarLintEngineImpl.java

```diff
@@ -224,6 +224,26 @@ public GlobalStorageUpdateCheckResult checkIfGlobalStorageNeedUpdate(ServerConfi
   }
 
   @Override
+  public StorageUpdateCheckResult checkIfModuleStorageNeedUpdate(ServerConfiguration serverConfig, String moduleKey, ProgressMonitor monitor) {
+    checkNotNull(serverConfig);
+    checkNotNull(moduleKey);
+    return withReadLock(() -> {
+      checkUpdateStatus();
+      ConnectedContainer connectedContainer = new ConnectedContainer(globalConfig, serverConfig);
+      try {
+        connectedContainer.startComponents();
+        return connectedContainer.checkForUpdate(moduleKey, new ProgressWrapper(monitor));
+      } finally {
+        try {
+          connectedContainer.stopComponents(false);
+        } catch (Exception e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1166)